src/message_handler.cc

```diff
@@ -128,6 +128,22 @@ void EmitSemanticHighlighting(QueryDatabase* db,
           continue;  // applies to for loop
         kind = func->def->kind;
         detailed_name = short_name;
+
+        // Check whether the function name is acturally there.
+        // If not, do not publish the semantic highlight.
+        std::string concise_name(
+            detailed_name.substr(0, detailed_name.find('<')));
+        auto start_line = sym.loc.range.start.line;
+        auto start_col = sym.loc.range.start.column;
+        if (0 <= start_line && start_line < working_file->index_lines.size()) {
```

Comment:
start_line >= 0 && start_line ...

Reply:
I don't remember whether I made hotfix to this code once or twice that `start_line` may go out of range.